unicorn/src/repository/zookeeper.cpp

```
@@ -0,0 +1,124 @@
+#include "zookeeper.hpp"
+
+#include <cocaine/context.hpp>
+#include <cocaine/logging.hpp>
+
+static int do_log(void* handler, const char* buf, size_t size) {
+    cocaine::logging::logger_t* log = static_cast<cocaine::logging::logger_t*>(handler);
+
+    std::string message(buf, size);
+    while (message.size() > 0 && message.back() == '\n') {
+        message.resize(message.size() - 1);
+    }
+
+    auto pos = message.find("ZOO_");
+    if (pos == std::string::npos || pos + 4 >= message.size()) {
```

BTW, why dont' you log such strange messages, but silently eat them?